Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/WinMDInfo.cs

```diff
@@ -0,0 +1,257 @@
+            foreach (KeyValuePair<BuildTarget, CompilationPlatformInfo> platformPair in platformDictionary)
+            {
+                // We presume exclude, then check
+                bool exclude = defaultExcludeValue;
+                foreach (string define in DefineConstraints)
+                {
+                    // Does this define exclude
+                    if (define.StartsWith("!"))
+                    {
+                        if (ContainsDefineHelper(define.Substring(1), inEditor, platformPair.Value))
+                        {
+                            exclude = true;
+                            break;
+                        }
+                    }
+                    else if (ContainsDefineHelper(define, inEditor, platformPair.Value))
+                    {
+                        // This platform is supported, but still search for !defineconstraitns that may force exclusion
+                        exclude = false;
+                    }
+                }
+
+                if (exclude)
+                {
+                    toExclude.Add(platformPair.Key);
+                }
+            }
+
+            foreach (BuildTarget buildTarget in toExclude)
+            {
+                platformDictionary.Remove(buildTarget);
+            }
+        }
+
+        private void TryAddEnabledPlatform(Dictionary<BuildTarget, CompilationPlatformInfo> playerPlatforms, Dictionary<string, bool> enabledPlatforms, string platformName, BuildTarget platformTarget)
+        {
+            if (enabledPlatforms.TryGetValue(platformName, out bool platformEnabled) && platformEnabled)
+            {
+                CompilationPlatformInfo platform = UnityProjectInfo.AvailablePlatforms.FirstOrDefault(t => t.BuildTarget == platformTarget);
+                if (platform != null)
+                {
+                    playerPlatforms.Add(platformTarget, platform);
+                }
+                else
+                {
+                    Debug.LogError($"Platform '{platformName}' was specified as enabled by '{ReferencePath.LocalPath}' plugin, but not available in processed compilation settings.");
+                }
+            }
+        }
+
+        private bool VerifyEnabledPlatforms(Dictionary<string, bool> enabledPlatforms)
+        {
+            bool valid = true;
+            foreach (KeyValuePair<string, bool> platform in enabledPlatforms)
+            {
+                if (platform.Value &&
+                    (platform.Key == "Win" ||
+                    platform.Key == "Win64" ||
+                    platform.Key == "iOS" ||
+                    platform.Key == "Android" ||
+                    platform.Key == "Editor"))
+                {
+                    Debug.LogError($"WinMDs should only be enabled for the WSA Player; however, {ReferencePath} was configured to support {platform.Key}.");
+                    valid = false;
+                    break;
+                }
```

Comment:
Do we want to correct this by just saying "we will only use it for UWP?" and logging an error, instead of just failing to use this as dependency?